From: Rolf Eike Beer <eike@sf-mail.de>
Date: Tue, 18 Aug 2015 10:14:43 +0000 (+0200)
Subject: FindPkgConfig: remove variable dereference
X-Git-Url: http://www.cmake.org/gitweb?p=cmake.git;a=commitdiff_plain;h=b9ec9392da21a3421e48c6961976060d872faffb

FindPkgConfig: remove variable dereference

If CMAKE_MINIMUM_REQUIRED_VERSION is not set because no
cmake_minimum_required() call is present this line can lead to an error as that
string is empty so too few arguments are passed to if():

See also "/var/tmp/paludis/build/kde-krdc-15.08.0/work/build/CMakeFiles/CMakeOutput.log".
See also "/var/tmp/paludis/build/kde-krdc-15.08.0/work/build/CMakeFiles/CMakeError.log".
CMake Error at /usr/share/cmake/Modules/FindPkgConfig.cmake:112 (elseif):
  given arguments:

    "VERSION_LESS" "3.1"

  Unknown arguments specified
Call Stack (most recent call first):
  /usr/share/cmake/Modules/FindPkgConfig.cmake:501 (_pkgconfig_parse_options)
  /usr/share/cmake/Modules/FindOpenSSL.cmake:43 (pkg_check_modules)
---

Index: cmake-3.2.2/Modules/FindPkgConfig.cmake
===================================================================
--- cmake-3.2.2.orig/Modules/FindPkgConfig.cmake
+++ cmake-3.2.2/Modules/FindPkgConfig.cmake
@@ -113,7 +113,7 @@ macro(_pkgconfig_parse_options _result _
       set(${_no_cmake_path} 1)
       set(${_no_cmake_environment_path} 1)
     endif()
-  elseif(${CMAKE_MINIMUM_REQUIRED_VERSION} VERSION_LESS 3.1)
+  elseif(CMAKE_MINIMUM_REQUIRED_VERSION VERSION_LESS 3.1)
     set(${_no_cmake_path} 1)
     set(${_no_cmake_environment_path} 1)
   endif()
